core/src/main/java/org/elasticsearch/search/suggest/SuggestionBuilder.java

```diff
@@ -212,6 +223,8 @@ protected void populateCommonFields(MapperService mapperService,
                 throw new IllegalArgumentException("Analyzer [" + luceneAnalyzer + "] doesn't exists");
             }
             suggestionContext.setAnalyzer(luceneAnalyzer);
+        } else {
+            suggestionContext.setAnalyzer(mapperService.searchAnalyzer());
         }
```

Comment:
So I took another look at how the analyzer gets resolved in PhraseSuggestParser currently, and there seem to be three potential steps that I tried to summarize [here](https://gist.github.com/cbuescher/39feab961fa7412aeb47). Now I wonder which of those make sense for the other two suggestion types, any which need to go into the SuggestionBuilder. Still undecided, but we will need to also adapt the base test to cover those cases. This should be done in another PR though. Since this addition here covers what was done in all three suggestion parsers in `verifySuggestion()`, I think it is good to do this here now.